server/pbench/bin/satools/extract_sa.py

```diff
@@ -0,0 +1,247 @@
+#!/usr/bin/env python3
+
+import os
+import re
+import sys
+import subprocess
+import configparser
+from satools import oscode
+from distutils.spawn import find_executable
+from optparse import OptionParser, make_option
+
+try:
+    from configparser import SafeConfigParser, NoSectionError, NoOptionError
+except:
+    from ConfigParser import SafeConfigParser, NoSectionError, NoOptionError
+
+
+nodename_pattern = re.compile(r'nodename=".*"')
+BASE_DIR = os.path.abspath(os.path.dirname('__file__'))
+DEFAULT_SADF_PATH = find_executable('sadf')
+
+# TODO:
+# check if DEFAULT_SADF_PATH is empty (covered under process_binary())
+
+# add unit tests for all cases of compatiblity.
+# Ones which also capture testing on TravisCI's platform distribution
+
+# Do we need a test to see if DEFAULT_SADF_PATH not present,
+# whereas binaries path contains the scripts needed to process?
+# Idk, since pbench-sysstat is installed de-facto.
```

Comment:
@portante should we be adding that test for `DEFAULT_SADF_PATH` in a separate PR or have this amended in the current one? The modifications done by you to `extract_sa.py` would end up being useful for https://github.com/distributed-system-analysis/sarjitsu 

Also, how do we plan on maintaining these 2 copies of `extract_sa.py`? (one in pbench, other in sarjitsu, and actually another in https://github.com/distributed-system-analysis/satools/blob/master/satools/extractsa.py). 

Related: https://github.com/distributed-system-analysis/satools/pull/1

thanks for the rework btw :-)) apologies for not being able to review this earlier. Working on my new year resolutions!